tests/undefined_symbol_in_sdfg_test.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python3
+# Copyright 2019-2024 ETH Zurich and the DaCe authors. All rights reserved.
+
+import pytest
+import numpy as np
+
+import dace
+from dace import symbolic
+
+
+def test_undefined_symbol_in_sdfg():
+    """Tests that an UndefinedSymbol can be used in an SDFG but raises an error when compiled."""
+    
+    @dace.program
+    def program_with_undefined_symbol(A: dace.float64[20, 20], B: dace.float64[20]):
+        # Create a transient with an undefined dimension
+        undefined_dim = symbolic.UndefinedSymbol()
+        tmp = dace.define_local([20, undefined_dim], dace.float64)
+        for i in range(20):
+            for j in range(20):
+                tmp[i, 0] = A[i, j]
+            B[i] = tmp[i, 0]
+    
+    sdfg = program_with_undefined_symbol.to_sdfg()
+    
+    # Validate SDFG - it should pass validation at this point
+    sdfg.validate()
+    
+    # Try to generate code - this should fail
+    with pytest.raises(TypeError, match="undefined symbol"):
+        sdfg.compile()
+
+
+def test_undefined_symbol_propagation():
```

Comment:
Why is this test in this file? The file name is `undefined_symbol_in_sdfg_test`, but the following test does not use an SDFG. Maybe it should go to the other test file (`tests/undefined_symbol_test.py`).